### PR TITLE
Update to use Kourier (and upgrade all dependency versions)

### PR DESCRIPTION
**Pull request by rick-a-lane-ii**
* Updates to use Kourier v1.1.0 (latest v1.2.0 requires Kubernetes v1.21)
  * (replaces Ambassador)
* Upgrades to Knative Serving v1.1.2 (latest v1.2.0 requires Kubernetes v1.21) 
* Upgrades to Tekton v0.32.1 (latest)
* Upgrades core Kubernetes versions to v0.22.5
  * We now support v1.20 and v1.21
  * (v1.19 is no longer supported based on dependencies)
  * (v1.22 and v1.23 would require additional work)
* Upgrades to controller-runtime v0.11.0 (latest) and controller-tools v0.8.0 (latest)
* Upgrades to go 1.17
* Upgrades k3s images for v1.20 and v1.21 to latest (for testing) and removes unsupported v1.19 version

### CODE REVIEW

File: pkg/operator/app/knativeservice.go [hunk 1]
```diff
@@ -20,20 +20,17 @@ import (
...
 )
 
 const (
-	AmbassadorID                         = "webhook"
 	KnativeServiceVisibilityClusterLocal = "cluster-local"
 )
 
 func ConfigureKnativeService(ctx context.Context, s *obj.KnativeService, wtd *WebhookTriggerDeps) error {
 	// FIXME This should be configurable
-	lifecycle.Annotate(ctx, s, AmbassadorIDAnnotation, AmbassadorID)
```
Comment (impl) [inline pkg/operator/app/knativeservice.go:36]: You can remove the `FIXME` comment here too.

File: pkg/util/testutil/kourier.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func doInstallKourier(ctx context.Context, cl client.Client, mapper meta.RESTMapper) error {
+	// Kourier requires Knative Serving; it won't detect it after the fact.
```
Comment (impl) [inline pkg/util/testutil/kourier.go:17]: I think this may only be true of Ambassador FWIW.

**Review by impl — APPROVED** [review]
LGTM!

File: pkg/util/testutil/kourier.go (hunk 2 above)
Comment (rick-a-lane-ii) [reply pkg/util/testutil/kourier.go:17]: Yeah, I figured both of those comments were probably no longer applicable, but I wasn't certain enough to remove them (yet). I don't think either comment are really necessary at this point, regardless, though, so I'll either remove them now or in the next update.

**Pull request merged**